plugins/org.yakindu.sct.generator.c/src/org/yakindu/sct/generator/c/CExpressionsGenerator.xtend

```diff
@@ -217,4 +214,20 @@ class CExpressionsGenerator extends ExpressionsGenerator {
 	def CharSequence ternaryGuard(Expression it) '''(«it.code») ? «TRUE_LITERAL» : «FALSE_LITERAL»'''
 	
 	override dispatch CharSequence code(FloatLiteral it) '''«value.toString»f'''
+	
+	def dispatch CharSequence code(FeatureCall it, OperationDefinition target) {
+		'''«target.access»(«IF target.eContainer.isMultiSM»«owner.handle»«ELSE»«scHandle»«ENDIF»«argCode»)'''
+	}
+	
+	protected def CharSequence argCode(List<Expression> expressions)
+		'''«FOR arg : expressions BEFORE ', ' SEPARATOR ', '»«arg.code»«ENDFOR»'''
+	
+	protected def CharSequence argCode(ElementReferenceExpression it) {
+		expressions.argCode
+	}
+	
+	protected def CharSequence argCode(FeatureCall it) {
+		expressions.argCode
+	}
```

Comment:
You can put all `argCode` methods together into:
```
protected def CharSequence argCode(ArgumentExpression it)
		'''«FOR arg : expressions BEFORE ', ' SEPARATOR ', '»«arg.code»«ENDFOR»'''
```

Reply:
you're totally right